lib/models/offline_recorder.dart

```diff
@@ -178,8 +171,44 @@ class OfflineRecorder {
     return News(
         title: decodedJson["title"],
         date: decodedJson["date"],
-        transcriptId: decodedJson["transcriptID"],
+        transcriptId: decodedJson["transcriptId"],
         audio: decodedJson["audio"],
+        fullTranscript: decodedJson["fullTranscript"],
         paragraphs: paragraphs);
   }
+
+  /// Gets the news of the given date, throws FileSystemException if the file containing the date's news doesn't exist
+  Future<News> loadNews(DateTime date) async {
+    // Sanitize inputs
+    String day = date.day < 10 ? "0${date.day}" : date.day.toString();
+    String month =
+        date.month < 10 ? "0${date.month.toString()}" : date.month.toString();
+    if (date.year < 1000) {
+      throw Exception("The date is too far in the past");
+    }
+
+    String filePath =
+        _appOfflineNewsPath + "${date.year}-$month-${day}_transcript.json";
+    if (!await File(filePath).exists()) {
+      throw FileSystemException("$filePath doesn't exist");
+    }
+
+    return _readNewsFile(File(filePath)).then((value) =>
+        value ??
+        (throw Exception("News for date ${date.toString()} doesn't exist")));
+  }
+
+  Future<List<News>> loadAllNews() async {
+    final dir = Directory(_appOfflineNewsPath);
+    final Iterable<File> files = (await dir.list().toList()).whereType<File>();
+
+    List<News> news = List.empty(growable: true);
+    for (var f in files) {
+      await _readNewsFile(f).then((n) => {
+            if (n != null) {news.add(n)}
+          });
+    }
+
+    return Future.value(news);

```

Comment:
```suggestion
    return news;
```

Can be simplified as the function is marked async.

Reply:
Yes indeed, thanks !